File: Bit Maniplulation/136. Single Number.py

### Bit Operation implementation ###
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        unordered_set<int> tmp;
        
        for (int i=0; i<nums.size)


### Hash map implementation ###
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        tmp = []
        
        for n in nums:
            if n not in tmp:
                tmp.append(n)
            else:
                tmp.remove(n)
        
        return tmp[0]

File: Bit Maniplulation/136. Single Number.cpp

// Bit Operation implementation
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int result=0;
        
        for (int i=0; i<nums.size(); i++) {
            result = nums[i] ^ result;
        }
        return result;
    }
};

/////////////////////////////////////////
// unordered_set implementation
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_set<int> tmp;
        
        for (int i=0; i<nums.size(); i++) {
            if (tmp.find(nums[i]) == tmp.end()) {
                tmp.insert(nums[i]);
            } else {
                tmp.erase(nums[i]);
            }
        }
        return *tmp.begin();
    }
};
